sql: clean up some uses of Unlock

RaduBerinde opened the pull request:
This is a minor cleanup change to use `defer.Unlock` when there are multiple exit paths that need to unlock.

Epic: none
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/107665)
<!-- Reviewable:end -->

maryliag reviewed (commented):
Thank you fixing this! Would you mind also updating [here](https://github.com/cockroachdb/cockroach/blob/master/pkg/sql/sqlstats/ssmemstorage/ss_mem_storage.go#L669) (since is also under sql package)?

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/107665#-:-NaIr0BOAvtUikZpaVoX:b-f4omtl)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @Santamaura, @smg260, @srosenberg, and @yuzefovich)



<!-- Sent from Reviewable.io -->

yuzefovich reviewed (commented):
> Reviewed 7 of 7 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/107665)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @RaduBerinde, @Santamaura, @smg260, and @srosenberg)

___
*[`pkg/sql/sqlliveness/slstorage/slstorage.go` line 232 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/107665#-NaIuyb02LI6cM-Zu6Q9:-NaIuyb02LI6cM-Zu6QA:b-zyywy) ([raw file](https://github.com/cockroachdb/cockroach/blob/dcfc41aa6c47a02606a7b8b2f53bdfd4b885d74b/pkg/sql/sqlliveness/slstorage/slstorage.go#L232)):*
> ```Go
> 
> 		// At this point, we know that the singleflight goroutine has been launched.
> 		// Releasing the lock here ensures that callers will either join the single-
> ```

nit: this comment might need an update given that we no longer release the lock right below.


<!-- Sent from Reviewable.io -->

RaduBerinde reviewed (commented):
Done

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/107665#-:-NaJE1-REHywLa0ePrYh:bb6un1j)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @Santamaura, @smg260, @srosenberg, and @yuzefovich)



<!-- Sent from Reviewable.io -->

maryliag reviewed (approved):
I will leave for other teams to check their code, but for the ones under `sql/sqlstats` <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 6 of 7 files at r1, 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/107665#-:-NaNaVzT12fPZvEk690n:b-wfriow)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @Santamaura, @smg260, and @srosenberg)



<!-- Sent from Reviewable.io -->

yuzefovich reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/107665#-:-NaNzLma5yr2FDVTGI0R:bnfp4nl)* status: :shipit: complete! 2 of 0 LGTMs obtained (waiting on @Santamaura, @smg260, and @srosenberg)



<!-- Sent from Reviewable.io -->

dhartunian reviewed (approved):
tracer.go and histogram.go <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> thank you!

> Reviewed 2 of 7 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/107665#-:-NaO-erq1H06M3AK8Vpn:baapykj)* status: :shipit: complete! 3 of 0 LGTMs obtained (waiting on @Santamaura, @smg260, and @srosenberg)



<!-- Sent from Reviewable.io -->

RaduBerinde reviewed (commented):
TFTRs!

Essential CI failure is a flake that already has an issue filed.

bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/107665#-:-NaO4I0n91nlRdNEDPNU:b-fizd9v)* status: :shipit: complete! 3 of 0 LGTMs obtained (waiting on @Santamaura, @smg260, and @srosenberg)



<!-- Sent from Reviewable.io -->

RaduBerinde committed:
sql: clean up some uses of Unlock

This is a minor cleanup change to use `defer.Unlock` when there are
multiple exit paths that need to unlock.

Epic: none
Release note: None

RaduBerinde commented:
bors r+
